Add name, url, image, and description to base context.

msporny opened the pull request:
This PR adds name, url, image, and description to the base context.

timothee-haudebourg commented on index.html:
```diff
@@ -155,9 +155,17 @@ <h3>Example</h3>
+  "expirationDate": "2028-11-15T12:00:00.0000000-06:00",
+  "name": "Utopia Driver's License",
+  "image": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUg...kSuQmCC",
+  "description": "Susan Turner's Utopia Driver's License.",
```
This next remark might be nitpicking: technically "Susan Turner" is the credential subject and "Susan Turner's Utopia Driver's License" is what's inside the `driversLicense` term. Outside of the credential subject is not the driver's license but an assertion by the license issuer that the subjects possesses a license. So I would maybe change that description or move it inside `driversLicense`. But again that's just nitpicking.

msporny replied:
Yes, what you say is technically accurate. 

The digital wallets that we've been working with use the `name`, `description`, and `image` properties to render previews of the VC in the wallet. The `description` property on the VC is used for a more detailed explanation about the VC. I was trying to provide some text that would be useful in the wallets that use the description property to render this sort of information.

timothee-haudebourg reviewed (changes requested):
You've added the terms in the `Iso18013DriversLicense` context, this means they will only make sense in a node of type `Iso18013DriversLicense` like this:
```json
{
  "@type": "Iso18013DriversLicense",
  "name": "Foo",
  "url": "https://example.org/foo",
  "image": "https://example.org/bar",
  "description": "Bar"
}
```
In particular, it won't work in the example you have changed.

msporny commented on context/v1.jsonld:
```diff
@@ -80,6 +86,7 @@
           "@type": "http://www.w3.org/2001/XMLSchema#base64Binary"
         },
         "un_distinguishing_sign": "https://w3id.org/vdl#unDistinguishingSign",
+        "url": "https://schema.org/url",
```
Right, I might default it to just `"@type": "@id"` to reduce schema.org weirdness (but still be compatible w/ schema.org). I think the typing information is a hold over from Microdata, so it's not necessary when using w/ RDF / JSON-LD. Unless you feel strongly about this, I'll make the change to ensure this is a URL (in the JSON-LD sense of the term).

timothee-haudebourg replied:
Unfortunately I think putting the weird schema.org type is necessary here if we want LD applications to interpret the documents correctly. `"@type": "@id"` is not the same as `"@type": "https://schema.org/URL"` and it is not correct w.r.t. schema.org. The resulting RDF dataset is not the same.

```json
{
  "@context": { "@type": "https://schema.org/URL" },
  "url": "http://example.org/foo"
}
```
is equivalent to:
```
_:subject <https://schema.org/url> "http://example.org/foo"^^<https://schema.org/URL> .
```

while
```json
{
  "@context": { "@type": "@id" },
  "url": "http://example.org/foo"
}
```
is equivalent to:
```
_:subject <https://schema.org/url> <http://example.org/foo> .
```
Here `<http://example.org/foo>` is not a schema.org URL, it is a resource identifier for something that may or may not be a schema.org URL. And if its an URL it may not be `http://example.org/foo` for instance if I write:
```json
{
  "@context": { "@type": "@id" },
  "url": "http://example.org/foo",
  "http://example.org/foo": { "@value": "http://example.org/completely/different/url", "@type": "https://schema.org/URL" }
}
```
it is equivalent to:
```
_:subject <https://schema.org/url> "http://example.org/completely/different/url"^^<https://schema.org/URL> .
```

timothee-haudebourg replied:
We might have the same issue with `https://schema.org/image`. Is the value of the `image` term supposed to be a reference to a `https://schema.org/ImageObject` (in which case `"@type": "@id"` is fine), or is it supposed to be an URL to the image (in which case `"@type": "https://schema.org/URL"` should be used).

msporny commented on context/v1.jsonld:
```diff
@@ -31,6 +31,7 @@
           "@type": "https://www.rfc-editor.org/rfc/rfc3339#full-date"
         },
         "birth_place": "https://w3id.org/vdl#birthPlace",
+        "description": "https://schema.org/description",
```
Ah, great catch! I meant to put it under `Iso18013DriversLicenseCredential`.

msporny commented on index.html:
```diff
@@ -155,9 +155,17 @@ <h3>Example</h3>
-  "expirationDate": "2022-08-27T12:00:00.0000000-06:00",
+  "issuer": {
+    "id": "did:key:z6MkjxvA4FNrQUhr8f7xhdQuP1VPzErkcnfxsRaU5oFgy2E5",
+    "name": "Utopia Department of Motor Vehicles",
```
Yes, good catch, I'll move these definitions under `Iso18013DriversLicenseCredential`.

msporny replied:
Fixed in fa8de475bd81c565ea758e730fdd9d2a64aeaba0

davidlehn commented on index.html:
```diff
@@ -167,8 +175,8 @@ <h3>Example</h3>
       "birth_date": "1998-08-28",
-      "issue_date": "2018-01-15T10:00:00.0000000-07:00",
-      "expiry_date": "2022-08-27T12:00:00.0000000-06:00",
+      "issue_date": "2023-01-15T10:00:00.0000000-07:00",
+      "expiry_date": "2028-08-27T12:00:00.0000000-06:00",
```
Seems these would be ok without sub-second precision:
```suggestion
      "issue_date": "2023-01-15T10:00:00-07:00",
      "expiry_date": "2028-08-27T12:00:00-06:00",
```

cobward reviewed (commented):
Could we instead just do something like:
```diff
 {
   "@context": {
     "@protected": true,
     "id": "@id",
     "type": "@type",
     "driversLicense": {
       "@id": "https://w3id.org/vdl#license",
       "@type": "@id"
     },
+    "name": "https://w3id.org/vdl#issuerName",
+    "url": "https://w3id.org/vdl#issuerUrl",
+    "description": "https://w3id.org/vdl#issuerDescription",
+    "image": "https://w3id.org/vdl#issuerImage",
     "Iso18013DriversLicenseCredential": "https://w3id.org/vdl#Iso18013DriversLicenseCredential",
 ```

msporny commented:
@timothee-haudebourg and @cobward we have made another pass through the vocabulary and context file ensuring the following:

* We moved the name, description, url, and image properties to be top-level defined so they're usable everywhere.
* We have properly scoped `driversLicense` under a `LicensedDriver` type.
* We have aligned the date-time values w/ XML Schema dateTime (to align w/ the VCWG direction on date and date-time values; we believe all ISO dates/times will be aligned with this datatype)
* We have applied a decision from a discussion earlier this week with Spruce wrt. image encoding formats (there was a new requirement to allow external URLs, data URLs, and the ability to understand the encoding format for the image from the data URL.
* We have fixed some of the example identifiers and descriptions to address @timothee-haudebourg's feedback and ensure the example is internally consistent wrt. Utopia.
* We confirmed that the context will work w/ the VC v1.1 context as well as the expected VC v2.0 context.

Going to merge this down now since we believe that everything has been addressed and you need to use it.
